keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/nuts/Tree.java

```diff
@@ -20,334 +23,369 @@
-        this.acceptedLeapfrogCount = acceptedLeapfrogCount;
-        this.shouldContinueFlag = shouldContinueFlag;
-        this.deltaLikelihoodOfLeapfrog = deltaLikelihoodOfLeapfrog;
-        this.treeSize = treeSize;
-    }
+    @Getter
+    private LeapfrogState forward;
 
     /**
-     * A helper method to create a basic tree
-     *
-     * @param position                 the starting position
-     * @param momentum                 the starting momentum
-     * @param gradient                 the starting gradient
-     * @param initialLogOfMasterP      the initial log prob
-     * @param sampleAtAcceptedPosition the initial sample
-     * @return a basic tree
+     * The backward most state in the tree
      */
-    public static Tree createInitialTree(Map<VariableReference, DoubleTensor> position,
-                                         Map<VariableReference, DoubleTensor> momentum,
-                                         Map<? extends
-                                             VariableReference, DoubleTensor> gradient,
-                                         double initialLogOfMasterP,
-                                         Map<VariableReference, ?> sampleAtAcceptedPosition) {
-
-        return new Tree(
-            new Leapfrog(position, momentum, gradient),
-            initialLogOfMasterP,
-            sampleAtAcceptedPosition,
-            STARTING_ACCEPTED_LEAPFROG,
-            STARTING_SHOULD_CONTINUE,
-            STARTING_DELTA_LIKELIHOOD,
-            STARTING_TREE_SIZE
-        );
+    @Getter
+    private LeapfrogState backward;
+
+    /**
+     * The accepted position in the tree
+     */
+    @Getter
+    private Proposal proposal;
+
+    /**
+     * Sum of momentum over all steps in tree
+     */
+    @Getter
+    private Map<VariableReference, DoubleTensor> sumMomentum;
+
+    /**
+     * The energy at the start state
+     */
+    @Getter
+    private final double startEnergy;
+
+    /**
+     * Maximum energy change on a step before the step is considered divergent
+     */
+    @Getter
+    private final double maxEnergyChange;
+
+    /**
+     * log sum of negative energy change
+     */
+    @Getter
+    private double logSumWeight;
+
+    /**
+     * A summation of the metropolis acceptance probability over each step in tree.
+     */
+    @Getter
+    private double sumMetropolisAcceptanceProbability;
+
+    /**
+     * Total number of steps in tree
+     */
+    @Getter
+    private int treeSize;
+
+    /**
+     * Height of binary tree as of last growth
+     */
+    @Getter
+    private int treeHeight;
+
+    /**
+     * True if the tree should continue to grow false otherwise. This will be false if the tree diverges or starts
+     * U-Turn
+     */
+    private boolean shouldContinueFlag;
+
+    /**
+     * @param startState                The leap frog for is initial step in the tree.
+     * @param proposal                  The current accepted position.
```

Comment:
Accepted comes form `accepted` as in Metropolis Hastings would accept it. It's the proposal because it's a Metropolis Hastings proposal that would be accepted.